2024 results (#134)
geeksblabla/stateofdev.ma: results/2024/sections/ai.mdx

---
title: AI
position: 6
---

# AI

> AI continues to be a hot topic in all the tech communities around the world. This year, we continue to explore its impact on Moroccan software developers and how they are adjusting to this evolving trend.

Reviewer: ```suggestion
AI continue to be a hot trend this year. Globally, some companies decided to fully embrace it, whereas other companies are much more reserved against its usage for their employees.
```

> AI continues to be a hot topic in all the tech communities around the world. This year, we continue to explore its impact on Moroccan software developers and how they are adjusting to this evolving trend.

Reviewer: ```suggestion
AI continues to be a hot topic in all the tech communities around the world. Globally, some companies decided to fully embrace it, whereas other companies are much more reserved against its usage by their employees. This year, we continue to explore its impact on Moroccan software developers and how they are adjusting to this evolving trend. 
```

Moroccan software developers are embracing the AI age. They are learning AI and building AI projects.

## Usage and Learning

> The majority of survey participants have integrated AI tools into their professional routines, with many of them relying on such technologies on a daily basis.

Reviewer: ```suggestion
The majority of survey participants have integrated AI tools into their professional routines, with many of them relying on such technologies on a daily basis much more compared to last year.
```

<Tabs>
  <TabItem title="2024">
    <Chart id="ai-q-0" title={false} year={2024} />
  </TabItem>
  <TabItem title="2023">
    <Chart id="ai-q-0" title={false} year={2023} />
  </TabItem>
</Tabs>

> A significant portion of respondents have already begun learning AI or even started building AI projects.

Reviewer: ```suggestion
Interest in AI keeps on growing. A significant portion of respondents have already begun learning AI or even started building AI projects.
```

<Tabs>
  <TabItem title="2024">
    <Chart id="ai-q-3" title={false} year={2024} />
  </TabItem>
  <TabItem title="2023">
    <Chart id="ai-q-3" title={false} year={2023} />
  </TabItem>
</Tabs>

ChatGPT and GitHub Copilot remain highly popular AI tools among Moroccan software developers.

<Tabs>
  <TabItem title="2024">
    <Chart id="ai-q-5" title={false} year={2024} />
  </TabItem>
  <TabItem title="2023">
    <Chart id="ai-q-5" title={false} year={2023} />
  </TabItem>
</Tabs>

> Most developers are using OpenAI's models for their AI projects.

Reviewer: ```suggestion
Amongst the respondents who use AI tools, most of them are using OpenAI's models for their AI projects.
```

<Tabs>
  <TabItem title="2024">
    <Chart id="ai-q-6" title={false} year={2024} />
  </TabItem>
  <TabItem title="2023">
    <Chart id="ai-q-6" title={false} year={2023} />
  </TabItem>
</Tabs>

> Companies are increasingly investing in using AI with more use cases in

Reviewer: ```suggestion
In the enterprise world, compagnies are slowly but increasingly investing in AI with more use cases in
```

production.

<Tabs>
  <TabItem title="2024">
    <Chart id="ai-q-4" title={false} year={2024} />
  </TabItem>
  <TabItem title="2023">
    <Chart id="ai-q-4" title={false} year={2023} />
  </TabItem>
</Tabs>

## Productivity and Future of AI

The majority of respondents feel that AI tools have increased their productivity.

<Tabs>
  <TabItem title="2024">
    <Chart id="ai-q-1" title={false} year={2024} />
  </TabItem>
  <TabItem title="2023">
    <Chart id="ai-q-1" title={false} year={2023} />
  </TabItem>
</Tabs>

> The majority of respondents believe that AI will have a significant impact on the future of software development.

Reviewer: ```suggestion
Impact-wise, the majority of respondents believes that AI will have some form of impact on the future of software develoment.
```
